pages/index.js

```diff
@@ -8,10 +11,122 @@ const StyledHeading = styled.h2`
 `;
 
 export default function HomePage({ goals, categoryColors }) {
+  const [isModalOpen, setIsModalOpen] = useState(false);
+  const [selectedGoal, setSelectedGoal] = useState(null);
+  const [newGoal, setNewGoal] = useState({
+    myGoal: "",
+    repetition: false,
+    targetPerInterval: 1,
+    interval: "day",
+    deadlineVisible: false,
+    deadline: "",
+    myNewGoals: [],
+  });
+
+  const handleOpenModal = () => {
+    setIsModalOpen(true);
+    setSelectedGoal(null);
+  };
+
+  const handleOpenModalFromListItem = (goal) => {
+    setIsModalOpen(true);
+    setSelectedGoal(goal);
+  };
+
+  const handleInputChange = (event) => {
+    const { name, value, type, checked } = event.target;
+
+    if (type === "checkbox") {
+      setNewGoal((prevState) => ({
+        ...prevState,
+        [name]: checked,
+      }));
+    } else {
+      setNewGoal((prevState) => ({
+        ...prevState,
+        [name]: value,
+        myGoal: value,
+      }));
+    }
+  };
+
+  const handleTargetPerIntervalChange = (value) => {
+    setNewGoal((prevState) => ({
+      ...prevState,
+      targetPerInterval: value,
+    }));
+  };
+
+  const handleAddGoal = (event) => {
+    event.preventDefault();
+    const myNewGoal = {
+      id: (goals.length += 1),
+      name: selectedGoal ? selectedGoal.description : newGoal.myGoal,
+      targetPerInterval: newGoal.targetPerInterval,
+      interval: newGoal.repetition ? newGoal.interval : null,
+      deadline: newGoal.deadlineVisible ? newGoal.deadline : null,
+      creationDate: new Date().toLocaleDateString(),
+    };
+    setNewGoal((prevGoals) => ({
+      myNewGoals: [...prevGoals.myNewGoals, myNewGoal],
+      myGoal: "",
+      repetition: false,
+      targetPerInterval: 1,
+      interval: "day",
+      deadlineVisible: false,
+      deadline: "",
+    }));
+    setIsModalOpen(false);
+    event.target.reset();
+    console.log([...newGoal.myNewGoals, myNewGoal]);
+    console.log(myNewGoal);
+  };
+
+  useEffect(() => {
+    const handleEscapeKeyPress = (e) => {
+      if (e.key === "Escape" && isModalOpen) {
+        closeModal();
+      }
+    };
+
+    document.addEventListener("keydown", handleEscapeKeyPress);
+
+    return () => {
+      document.removeEventListener("keydown", handleEscapeKeyPress);
+    };
```

Comment:
very cool, that you build that in! 😊
I think you don't need the return here 
```suggestion
  useEffect(() => {
    const handleEscapeKeyPress = (e) => {
      if (e.key === "Escape" && isModalOpen) {
        closeModal();
      }
    };

    document.addEventListener("keydown", handleEscapeKeyPress);
```

Reply:
☝️ the return here is very good. It cleans up the effect by removing the event listener that was added when the effect is attached. Please keep it :)

Reply:
https://react.dev/reference/react/useEffect#connecting-to-an-external-system